include/net/nrf_cloud_alerts.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef NRF_CLOUD_ALERTS_H_
+#define NRF_CLOUD_ALERTS_H_
+
+/** @file nrf_cloud_alerts.h
+ * @brief Module to provide mechanism to send application alerts to nRF Cloud.
+ */
+
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @defgroup nrf_cloud_alerts nRF Cloud Alerts
+ * @{
+ */
+
+/** @brief Alert type */
+enum nrf_cloud_alert_type {
+	/** Placeholder; not meant to be sent. */
+	ALERT_TYPE_NONE,
+	/** Device finished booting. */
+	ALERT_TYPE_DEVICE_NOW_ONLINE,
+	/** Device is shutting down. */
+	ALERT_TYPE_DEVICE_GOING_OFFLINE,
+	/** Generic message (see description field). */
+	ALERT_TYPE_MSG,
+	/** Operating temperature outside of limits. */
+	ALERT_TYPE_TEMPERATURE,
+	/** Battery level too low. */
+	ALERT_TYPE_BATTERY,
```

Comment:
`ALERT_TYPE_WDT_RESET` could be useful.
or `ALERT_TYPE_DEVICE_NOW_ONLINE_WDT` to align with existing.